disdat/utility/aws_s3.py

```diff
@@ -585,6 +593,21 @@ def get_s3_resource():
     return b3.resource('s3')
 
 
+# Module global variable used to initialize and store process state
+s3_resource = None
```

Comment:
Maybe move the global variable to the top of the file along with this function.    One thing I like about the per-invocation acquisition of the s3_resource is that it's simple and tidy.   Now if we add any S3 code to this module, we need to know that we should use this global (and also not re-assign it and break other code).